yt/funcs.py

```diff
@@ -994,11 +994,18 @@ def enable_plugins():
                 os.path.join(old_config_dir, my_plugin_name),
                 os.path.join(CONFIG_DIR, my_plugin_name))
         mylog.info("Loading plugins from %s", _fn)
-        execdict = yt.__dict__.copy()
+        ytdict = yt.__dict__
+        execdict = ytdict
         execdict['add_field'] = my_plugins_fields.add_field
+        localdict = {}
         with open(_fn) as f:
             code = compile(f.read(), _fn, 'exec')
-            exec(code, execdict)
+            exec(code, execdict, localdict)
+        ytnamespace = list(ytdict.keys())
+        for k in localdict.keys():
+            if k not in ytnamespace:
+                if callable(localdict[k]):
```

Comment:
I recognize the name here is to add functions, but I am not totally clear on why we need to do this if the definition is callable -- is it because the functions don't get redefined as locally-bound variables?  Do we need to do this for any other types?

Reply:
It's because the docs for the plugins file says that only functions defined in the plugins file will be available in the yt namespace, not any technical issue. I think we just need to check if the object is callable or not.